### PR TITLE
Managing publisher connections with poolboy

### PR DESCRIPTION
**Pull request by mariaclrd**
At the moment Task bunny opens a new channel every time to publish a message. Using poolboy we will be able to reuse the existing channels. New channels will be opened only when needed.

### CODE REVIEW

File: lib/task_bunny/publisher.ex [hunk 1]
```diff
@@ -21,8 +21,11 @@ defmodule TaskBunny.Publisher do
     e in [ConnectError, PublishError] -> {:error, e}
   end
 
+  @timeout 10_000
```
Comment (ono) [inline lib/task_bunny/publisher.ex]: can you rename this to poolboy_timeout and move this under alias definition on top of the file?

File: lib/task_bunny/publisher.ex [hunk 2]
```diff
@@ -31,15 +34,16 @@ defmodule TaskBunny.Publisher do
...
 
-    with {:ok, channel} <- AMQP.Channel.open(conn),
-         :ok <- AMQP.Basic.publish(channel, exchange, routing_key, message, options),
-         :ok <- AMQP.Channel.close(channel) do
+    with :ok <-
+           :poolboy.transaction(
+             :publisher,
+             &GenServer.call(&1, {:publish, host, exchange, routing_key, message, options}),
+             @timeout
+           ) do
```
Comment (ono) [inline lib/task_bunny/publisher.ex]: It now doesn't really make sense to use with clause here. Can you change it to case or if?

File: lib/task_bunny/publisher_worker.ex [hunk 3]
```diff
@@ -0,0 +1,55 @@
+defmodule TaskBunny.PublisherWorker do
+  @moduledoc """
+    GenServer worker to publish a message on a queue
```
Comment (ono) [inline lib/task_bunny/publisher_worker.ex]: Unnecessary indentation here and all other `@doc`

**Comment by mariaclrd** [comment]
Thank you for your comments! :) They have been addressed in the latest commit

**Comment by kalymero** [comment]
LGTM!

File: lib/task_bunny/config.ex [hunk 4]
```diff
@@ -215,4 +215,26 @@ defmodule TaskBunny.Config do
       _ -> [TaskBunny.FailureBackend.Logger]
     end
   end
+
+  @doc """
+  Returns the publisher pool size for poolboy. 15 by default
+  """
+  @spec publisher_pool_size :: integer
+  def publisher_pool_size do
+    case Application.fetch_env(:task_bunny, :publisher_pool_size) do
```
Comment (kalymero) [inline lib/task_bunny/config.ex]: ~~ i think you could use `Application.get_env/3` to specify a default value

**Pull request merged**